sharedbuffers/mapped_struct.py

```diff
@@ -132,19 +132,28 @@
     long = int
     basestring = (bytes, str)
 
+_memoryview = cython.declare(object)
+_memoryview = memoryview
+
 @cython.cfunc
 @cython.inline
 def make_memoryview(a):
     if python3:
-        return memoryview(a).cast("B")
+        if type(a) is not _memoryview:
+            a = _memoryview(a)
+        if a.format != 'B' or a.ndim != 1:
+            a = a.cast('B')
+        return a
     else:
         return buffer(a)
 
 if cython.compiled:
     # Compatibility fix for cython >= 0.23, which no longer supports "buffer" as a built-in type
+    buffer = cython.declare(object, buffer)  # lint:ok
```

Comment:
I understand the `cython.declare(object` but why initialize it to "itself"?

Reply:
I just want a native global pointer to `buffer`, and I'm calling it `buffer`

Reply:
I mean, you could just do `buffer = cython.declare(object)`, you are setting it right below this